src/chainstate/stacks/boot/bns.clar

```diff
@@ -0,0 +1,973 @@
+(define-constant NAME_PREORDER_CLAIMABILITY_TTL u10)
+(define-constant NAME_GRACE_PERIOD_DURATION u5)
+
+(define-constant ATTACHMENTS_INV_PAGE_SIZE u8)
+(define-data-var attachments-inv-index-cursor uint u0)
+(define-data-var attachments-inv-page-cursor uint u0)
+(define-map attachments-inv 
+    ((page uint) (index uint)) 
+    ((content-hash (buff 20))))
+
+;; Price tables
+(define-constant NAMESPACE_PRICE_TIERS (list
+  u96000 
+  u9600 u9600 
+  u960 u960 u960 u960 
+  u96 u96 u96 u96 u96 u96 u96 u96 u96 u96 u96 u96 u96))
+
+;;;; Data
+(define-map namespaces
+  ((namespace (buff 19)))
+  ((namespace-import principal)
+   (revealed-at uint)
+   (launched-at (optional uint))
+   (namespace-version uint)
+   (renewal-rule uint)
+   (price-function (tuple 
+    (buckets (list 16 uint)) 
+    (base uint) 
+    (coeff uint) 
+    (nonalpha-discount uint) 
+    (no-vowel-discount uint)))))
+
+(define-map namespace-preorders
+  ((hashed-salted-namespace (buff 20)) (buyer principal))
+  ((created-at uint) (claimed bool) (stx-burned uint)))
+
+(define-non-fungible-token names (tuple (name (buff 16)) (namespace (buff 19))))
+
+(define-map owner-name ((owner principal)) ((name (buff 16)) (namespace (buff 19))))
+
+(define-map name-properties
+  ((name (buff 16)) (namespace (buff 19)))
+  ((registered-at (optional uint))
+   (imported-at (optional uint))
+   (revoked-at (optional uint))
+   (zonefile-hash (buff 20))))
+
+(define-map name-preorders
+  ((hashed-salted-fqn (buff 20)) (buyer principal))
+  ((created-at uint) (claimed bool) (stx-burned uint)))
+
+(define-private (min (a uint) (b uint))
+  (if (<= a b) a b))
+
+(define-private (max (a uint) (b uint))
+  (if (> a b) a b))
+
+(define-read-only (compute-namespace-price? (namespace (buff 19)))
+  (let ((namespace-len (len namespace)))
+    (asserts!
+      (> namespace-len u0)
+      (err ERR_NAMESPACE_BLANK))
+    (ok (get value (fold 
+      element-at 
+      NAMESPACE_PRICE_TIERS 
+      (tuple (limit (min u8 namespace-len)) (cursor u0) (value u0)))))))
+
+(define-private (element-at (i uint) (acc (tuple (limit uint) (cursor uint) (value uint))))
+  (if (is-eq (get cursor acc) (get limit acc))
+    (tuple (limit (get limit acc)) (cursor (+ u1 (get cursor acc))) (value i))
+    (tuple (limit (get limit acc)) (cursor (+ u1 (get cursor acc))) (value (get value acc)))))
+  
+(define-private (get-exp-at-index (buckets (list 16 uint)) (index uint))
+  (get value (fold element-at buckets (tuple (limit index) (cursor u0) (value u0)))))
+
+(define-private (is-digit (char (buff 1)))
+  (or 
+    (is-eq char 0x30) ;; 0
+    (is-eq char 0x31) ;; 1
+    (is-eq char 0x32) ;; 2
+    (is-eq char 0x33) ;; 3
+    (is-eq char 0x34) ;; 4
+    (is-eq char 0x35) ;; 5
+    (is-eq char 0x36) ;; 6
+    (is-eq char 0x37) ;; 7
+    (is-eq char 0x38) ;; 8
+    (is-eq char 0x39))) ;; 9
+
+(define-private (is-lowercase-alpha (char (buff 1)))
+  (or 
+    (is-eq char 0x61) ;; a
```

Comment:
Same here

Reply:
This is unfortunately not something you can do with Clarity - issue tracked here https://github.com/blockstack/stacks-blockchain/issues/1172